numpy/core/include/numpy/ufuncobject.h

```diff
@@ -130,8 +130,8 @@ typedef struct _tagPyUFuncObject {
         /* The number of elements in 'functions' and 'data' */
         int ntypes;
 
-        /* Used to be unused field 'check_return' */
-        int reserved1;
+        /* Used to be unused field 'check_return', repurposed in 1.16 */
+        int version;
```

Comment:
Shouldn't this be a constant?  `const int version;`

Reply:
We use `const int` for pickle version types in NumPy source for example

Reply:
`const` has no meaning in `c`, it is a `c++` construct

Reply:
What? `const int x` is meaningful in C--in fact, it is not even the same type as `int x`. It has meaning to the compiler and if you try assigning to it later in the control flow it won't work.

Also, you can use `const` with pointers too, in a different sense, plenty of uses for `const` in C.

Here, I think it is pretty clear that the code shouldn't be re-assigning to the constant `version`, so it should be typed as a `const int`, not `int`

Reply:
correct. Fixing.

Reply:
Grepping in the code base shows this (and a few others):

`numpy/core/src/multiarray/methods.c:    const int version = 1;`

I guess we may have to be cautious because we're using a really old C-standard in the CI tests, and `const` didn't exist in the K&R / C89 days I think. But it is clearly in our source. Apparently you can also do `int const version = 1;` and some prefer that but our code base does the first one. Is that not working?

Reply:
Here the `version` is a field in a struct. While I am a bit ambivalent about the desire to make it `const`, I have added code to do this in 31a0d21d7.